### PR TITLE
Upgrade to checkstyle maven plugin 3.1.1

### PR DESCRIPTION
**Pull request by aaime**
Updates to latest checkstyle plugin. Since last update, it deprecated and removed some checks, and made javadoc tests stricter (the settings removed were used to make it more lax). Basically had to remove all empty javadoc param/return/throws tags.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Committs changing the REST API, or any configuration object, should check it the REST API docs (Swagger YAML files and classic documentation) need to be updated.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by bradh** [comment]
Possibly https://checkstyle.sourceforge.io/config_javadoc.html#NonEmptyAtclauseDescription might have been enough, but since you've already done the work, probably better cleaned up.

File: src/community/dyndimension/src/main/java/org/geoserver/wms/dimension/DynamicDefaultValueSelectionFactory.java [hunk 1]
```diff
@@ -662,14 +653,7 @@ private FeatureCalc getFeatureCalcForStrategy(
...
 
-    /**
-     * Applies the normal policy, but restricted to the
-     *
-     * @param resource
-     * @param config
-     * @param restrictions
-     * @throws IOException
-     */
+    /** Applies the normal policy, but restricted to the */
```
Comment (bradh) [inline src/community/dyndimension/src/main/java/org/geoserver/wms/dimension/DynamicDefaultValueSelectionFactory.java]: Don't think this is adding anything.

File: src/community/json-ld/src/main/java/org/geoserver/jsonld/configuration/JsonLdTemplate.java [hunk 2]
```diff
@@ -35,11 +35,7 @@ public JsonLdTemplate(Resource templateFile, NamespaceSupport namespaces) {
         }
     }
 
-    /**
-     * Check if json-ld template file has benn modified an eventually reload it.
-     *
-     * @return
-     */
+    /** Check if json-ld template file has benn modified an eventually reload it. */
```
Comment (bradh) [inline src/community/json-ld/src/main/java/org/geoserver/jsonld/configuration/JsonLdTemplate.java]: Typos. `benn` -> `been` and `an` -> `and`

File: src/community/ogcapi/ogcapi-changeset/src/main/java/org/geoserver/api/changeset/ModifiedTiles.java [hunk 3]
```diff
@@ -186,18 +186,12 @@ private GridSubset toGridSubset(
...
     public Iterator<long[]> getTiles() {
         return new TileIterator(subsets, zoomStart, zoomEnd);
     }
 
-    /**
-     * Retruns the number of tiles modified by the list of changes
-     *
-     * @return
-     */
+    /** Retruns the number of tiles modified by the list of changes */
```
Comment (bradh) [inline src/community/ogcapi/ogcapi-changeset/src/main/java/org/geoserver/api/changeset/ModifiedTiles.java]: Returns

File: src/community/qos/src/main/java/org/geoserver/qos/BaseQosXmlEncoder.java [hunk 4]
```diff
@@ -254,12 +250,7 @@ public void encodeOwsRange(Translator tx, String min, String max) {
         tx.end(owsPrefix() + ":" + OWS_RANGE_TAG);
     }
 
-    /**
-     * <ows:Value>Vesienhoitoalueet</ows:Value>
-     *
-     * @param tx
-     * @param value
-     */
+    /** <ows:Value>Vesienhoitoalueet</ows:Value> */
```
Comment (bradh) [inline src/community/qos/src/main/java/org/geoserver/qos/BaseQosXmlEncoder.java]: Oh.

File: src/community/wps-remote/src/main/java/org/geoserver/wps/remote/RemoteProcessClient.java [hunk 5]
```diff
@@ -96,11 +96,7 @@
     /** */
     protected String certificatePassword = null;
 
-    /**
-     * The default Cosntructor
-     *
-     * @param remoteProcessFactory
-     */
+    /** The default Cosntructor */
```
Comment (bradh) [inline src/community/wps-remote/src/main/java/org/geoserver/wps/remote/RemoteProcessClient.java]: Typo.

**Comment by aaime** [comment]
@bradh I don't have time to go through and fix existing javadoc, but if you want to have a go at it, you'd be more than welcomed, there is indeed a lot that can be improved without having to understand the actual code being documented. @ianturton did a run of spell checking on GeoTools, not sure what tools were used (if any?).

**Comment by bradh** [comment]
Per my comment, concur not important to resolve the typos etc.

**Pull request merged**